lib/executor/externally_controlled.go

```diff
@@ -479,6 +479,7 @@ func (rs *externallyControlledRunState) handleConfigChange(oldCfg, newCfg Extern
 				executionState.ReturnVU(rs.vuHandles[i].initVU, false)
 			}
 			rs.vuHandles[i] = nil
+			executionState.ModInitializedVUsCount(-1)
```

Comment:
Was *this* the actual issue of not scaling down ?

Reply:
Yes :D `initializedVUs` wasn't decremented anywhere. Though not sure if this is the right approach or place for it :-/

Reply:
This should only be done when we're not returning the VU back to the buffer pool though, like this:
```
			if i < rs.startMaxVUs {
				// return the initial planned VUs to the common buffer
				executionState.ReturnVU(rs.vuHandles[i].initVU, false)
			} else {
				executionState.ModInitializedVUsCount(-1)
			}

```

Reply:
To demonstrate, use the following options:
```js
export let options = {
    scenarios: {
        extc1: {
            executor: "externally-controlled",
            vus: 5,
            maxVUs: 10,
            duration: "10s",
        },
        const_loop_vus: {
            executor: "constant-vus",
            vus: 10,
            duration: "20s",
            startTime: "10s",
            gracefulStop: "0s",
        },
    },
};
```

see the `00/00 VUs` in the description here:
```
running (14.1s), 00/00 VUs, 139 complete and 15 interrupted iterations
extc1          ✓ [======================================] 05/10 VUs  10s   
const_loop_vus ✗ [======>-------------------------------] 10 VUs     04.1s/20s
```